robocorp-code/src/robocorp_code/inspector/java/robocorp_java/_inspector.py

```diff
@@ -0,0 +1,92 @@
+from typing import List, Optional, TypedDict, Union
+
+from JABWrapper.context_tree import ContextNode, ContextTree
+from JABWrapper.jab_wrapper import JavaAccessBridgeWrapper, JavaWindow
+
+ColletedTreeTypedDict = TypedDict(
+    "ColletedTreeTypedDict",
+    {
+        "matches": Union[ContextNode, List[ContextNode]],
+        "tree": ContextNode,
+    },
+)
+
+
+class ElementInspector:
+    def __init__(self) -> None:
+        self._context: Optional[ContextNode] = None
+
+    def _start_event_pump(func, *args, **kwargs):
+        def wrapper(self: "ElementInspector", *args, **kwargs):
```

Comment:
The way the code is structured, the `JavaAccessBridgeWrapper` is created in a thread and an event loop is continually run in that thread. i.e.:

```
            if PeekMessage(message, 0, 0, 0, REMOVE_FROM_QUEUE):
                TranslateMessage(message)
                log.debug("Dispatching msg={}".format(repr(message)))
                DispatchMessage(message)
                return True
```

Now, although the `JavaAccessBridgeWrapper` was created in that thread, afterwards it's not really used in that thread, it's only used in this thread... 

Given that, my questions are the following: 
1. why create the `JavaAccessBridgeWrapper` in that thread in the first place? (i.e.: why not create it in this thread as that thread will just keep running that loop)
2. Isn't it dangerous having that loop running in a thread while having this thread manipulate the `JavaAccessBridgeWrapper`? (i.e.: usually UI's are single-threaded and accessing things from multiple threads could have bad repercussions)

Reply:
Jab wrapper needs to be part of the thread that pumps the window events, otherwise the API would stall and never return anything. Also we cannot call the API in that thread as we need to pump the events for the API to return results.

Reply:
Ok, (seems fair that the `JavaAccessBridgeWrapper` is in the same thread that's running the event loop). I guess everything is Ok then if the APIs used from this thread are thread safe :)